### DISCUSSION TITLE
rake webでimg srcがローカルのフルパスになるが相対パスにしたい

### DISCUSSION THREAD

**Discussion by Nyoho** [kmuto/review#1894] [Q&A]
rake web で `webroot/` 以下にウェブページを生成しています。

`@<icon>` や `//image...` で入れ込んだ画像の `img` タグの `src` の値が、ローカルファイルシステムのフルパスになっていました。

その img src を `webroot/` からの相対パスになるようにしたいのですが、どのようにするとできるでしょうか。

## 環境

- Re:VIEW 5.7.0
- ruby 3.2.2
- macOS Ventura 13.3.1 (a)

**Comment by kdmsnr** [comment]
本当ですね。いちおう以下のようにすると相対パスになるっぽいですが、適切かどうかはわかりません。

```diff
--- a/lib/review/webmaker.rb
+++ b/lib/review/webmaker.rb
@@ -101,7 +101,7 @@ module ReVIEW
     end
 
     def generate_html_files(yamlfile)
-      @basedir = File.absolute_path(File.dirname(yamlfile))
+      @basedir = File.path(File.dirname(yamlfile))
       @path = build_path
       remove_old_files(@path)
       Dir.mkdir(@path)
```

**Reply by Nyoho** [reply]
ありがとうございます。これで相対パスにできました。
rake webは出版には関係ない、手元での確認用なのかも知れませんが、絶対パスにしておく理由がないような気がします。PRで修正することを前提にissueにしてみてもいいでしょうか。

**Reply by Nyoho** [reply]
もう速攻でされていたことに今頃気付きました。すみません。< pr
